main.py

```diff
@@ -0,0 +1,253 @@
+    point2_1, point2_2 = find_intersection_points(theta2, d2, image.shape)
+
+    if need_plot is True:
+        plt.figure()
+        plt.title('Finding door width')
+        plt.imshow(image, cmap="gray")
+        plt.plot((point1_1[0], point1_2[0]), (point1_1[1], point1_2[1]), '-r')
+        plt.plot((point2_1[0], point2_2[0]), (point2_1[1], point2_2[1]), '-r', label='Detected lines')
+        plt.legend(loc='upper right')
+
+    # calculate the "average" distance between this two lines, it will be treated as the door width
+    door_width = calculate_avg_door_width(point1_1, point2_2, theta1, theta2, d1, d2, need_plot)
+    return door_width, (theta1, d1), (theta2, d2)
+
+
+def my_binarization(image, th, color):
+    bin_img = [[1 if np.math.dist(pixel, color) < th else 0 for pixel in line] for line in image]
+    return bin_img
+
+
+def get_two_largest_components(segmentation):
+    labels = label(segmentation)
+    props = regionprops(labels)
+    areas = [prop.area for prop in props]
+    if len(areas) == 0:
+        print("No components found")
+        return None, None
+    if len(areas) == 1:
+        print("Single component found")
+        return labels, None
+    sorted_areas = sorted(enumerate(areas), key=lambda p: p[1], reverse=True)
+    largest_comp_id = sorted_areas[0][0]
+    second_largest_comp_id = sorted_areas[1][0]
+    return labels == (largest_comp_id + 1), labels == (second_largest_comp_id + 1)
+
+
+def check_fit_in(segment, door, extension_cff=0.25):
+    width, (theta1, d1), (theta2, d2) = door
+    not_fit_in = np.zeros_like(segment, dtype=bool)
+    result = True
+
+    for y in range(len(segment)):
+        for x in range(len(segment[y])):
+            if segment[y][x]:
+                signed_dist1 = y * np.sin(theta1) + x * np.cos(theta1) - d1
+                signed_dist2 = y * np.sin(theta2) + x * np.cos(theta2) - d2
+                if signed_dist1 * signed_dist2 > 0.0:
+                    if min(abs(signed_dist1), abs(signed_dist2)) > width * extension_cff:
+                        not_fit_in[y][x] = True
+                        result = False
+
+    return result, not_fit_in
+
+
+resize_cff = 0.25
+brown = [0.22647059, 0.12352941, 0.07352941]
+plot_params = {'initial image': False, 'all th': False, 'local th': False, 'otsu + my binary': False,
+               'edges maps': False, 'door lines': True, 'result segmentation': True}
+
+img = io.imread("Photoset/1_t.JPG")
+initial_h, initial_w, channels = img.shape
+img_resized = resize(img, (initial_h * resize_cff, initial_w * resize_cff))
+img_gray = rgb2gray(img_resized)
+img_gray_blur = gaussian(img_gray, 2)
+if plot_params['initial image']:
+    plt.imshow(img_resized)
+    plt.title('Initial image')
+
+if plot_params['all th']:
+    fig_all_th, ax_all_th = try_all_threshold(img_gray_blur, figsize=(15, 15), verbose=False)
+
+img_local = img_gray_blur >= threshold_local(img_gray_blur, 5, method='gaussian', param=25)
+if plot_params['local th']:
```

Comment:
I think each 'if' code block can be incapsulated in new function to look more convinient.